pandas/io/json/_json.py

```diff
@@ -850,6 +850,26 @@ def _convert_axes(self):
         obj = self.obj
         assert obj is not None  # for mypy
         for axis_name in obj._AXIS_ORDERS:
+            if (
+                self.orient == "split"
+                and axis_name == "index"
+                and obj._get_axis(axis_name).dtype == "object"
+                and obj.index.inferred_type == "string"
+            ):
+                index_list = list(obj._get_axis(axis_name).values)
+                are_all_strings = all(type(element) == str for element in index_list)
```

Comment:
we have helpers got these kinds of checks in `pandas/core/dtypes/common.py`, may as well use those